### PR TITLE
Add DDP multi-GPU training support

### PR DESCRIPTION
**Pull request by sreevadde**
## Summary

This PR adds DistributedDataParallel (DDP) support for efficient multi-GPU training, which provides better scaling compared to the existing DataParallel approach.

### Changes

- **train_ddp.py**: New training script using `torch.distributed` with NCCL backend
- **train_one_epoch_ddp**: DDP-aware training loop with proper rank handling
- **snap_v4.yaml**: Example config for 3-class action detection

### Usage

```bash
torchrun --nproc_per_node=4 train_ddp.py configs/your_config.yaml
```

### Why DDP over DataParallel?

- **Better scaling**: Each GPU runs its own process, avoiding GIL contention
- **Memory efficiency**: Gradient synchronization is more efficient
- **Compatibility**: Works with models that receive list/dict inputs (DataParallel struggles with non-tensor inputs)

### Testing

Tested with 4x NVIDIA L4 GPUs - achieves ~4x speedup compared to single GPU training.

### CODE REVIEW

**Comment by sreevadde** [comment]
Closing - will submit a cleaner PR with only general DDP improvements
